### PR TITLE
Update so newlines will be filtered from logging

### LINKED ISSUE #243: Protect against log forging
Issue Description
--
Static scanning of client code revealed a log forging vulnerability in our code relating to logging parameters received from an external source. Looking in BaseRestController it seems FUSE is exposed to this too (as we log the requestId, which is a string).

This obviously isn't a critical issue, but it is good to be aware of these types of issues.

See https://www.baeldung.com/jvm-log-forging for details.

To add more colour, this issue only came up after we delomboked the code before running the scan.

Design
--
[Description of the root cause for the issue and how to resolve it. Should be complete before Review is approved]

Documentation Changes
--
[Description of changes necessary to the Fuse documentation due to the changes in response to this issue. Should be complete before Review is approved]

Test Evidence
--
Before:
image

After:
image

Validation in Develop
-- 
[Proof issue is resolved in Develop. Should be complete before Issue is closed]

### PR DESCRIPTION
**Pull request by GalateaTom**
Change does similar job to https://github.com/galatea-associates/fuse-starter-java/pull/244
Unlike @Gala-Sortino 's change this method will not filter out \t but it will still filter out newlines.  It's more concise and should be safer as it runs on every log line, not just those when we call encode.
Change been tested on hardcoded logs to prove that newlines are escaped, although I have not been able to test this against input actually provided to the RESTful service because those characters are escaped before they get to the logger (so newlines weren't appearing in logs even before I made this change).
Happy to discuss what best solution is.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -58,6 +59,37 @@
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-log4j2</artifactId>
+			<exclusions>
+				<exclusion>
+					<groupId>org.apache.logging.log4j</groupId>
+					<artifactId>log4j-slf4j-impl</artifactId>
+				</exclusion>
```
Comment (GalateaFarhaad) [inline pom.xml]: I don't think it's necessary to exclude these transitive dependencies. Adding them as explicit dependencies with explicit versions should be sufficient to override the transitive versions.

**Comment by GalateaTom** [comment]
I've run few tests using Postman and confirm that using that we can pass in line breaks and they are not included in the logs:
Logging Line Breaks

File: pom.xml (hunk 1 above)
Comment (GalateaTom) [reply pom.xml]: Ok, I was wondering if this was necessary or not.  Had a read through http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html and you're right, it's enough that we're stating the versions later on.

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java [hunk 2]
```diff
@@ -16,9 +18,21 @@
...
+    //Proper fix requires version 2.10.0 or higher of the log4j dependencies
+    //See Issue #243 for more information.
     if (requestId != null) {
-      log.info("Request received with id: {}", requestId);
-      Tracer.setExternalRequestId(requestId);
+      String cleanedRequestId;
+      try {
+        cleanedRequestId = UriUtils.encode(requestId, "ISO-8859-1");
+        log.info("Request received.  Cleaned id: {}", cleanedRequestId);
+      } catch (UnsupportedEncodingException e) {
```
Comment (GalateaEric) [inline src/main/java/org/galatea/starter/entrypoint/BaseRestController.java]: have a look at: https://projectlombok.org/features/SneakyThrows for this case.  In particular have a look at the 2nd "common use case".

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java [hunk 3]
```diff
@@ -16,9 +18,21 @@
...
+    //This is a temporary solution for cleaning special characters from the REST request params
+    //It will not fix newlines in the parameters or JSON body being printed by the @Log annotation
+    //Proper fix requires version 2.10.0 or higher of the log4j dependencies
+    //See Issue #243 for more information.
     if (requestId != null) {
-      log.info("Request received with id: {}", requestId);
-      Tracer.setExternalRequestId(requestId);
+      String cleanedRequestId;
+      try {
+        cleanedRequestId = UriUtils.encode(requestId, "ISO-8859-1");
```
Comment (GalateaEric) [inline src/main/java/org/galatea/starter/entrypoint/BaseRestController.java]: let's stick with just UTF-8 for now.  UTF-8 apparently must be available in the JVM.

Further reading: https://www.garron.me/en/linux/charset-encoding-html-change.html

**Review by GalateaEric — CHANGES_REQUESTED** [review]

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java (hunk 2 above)
Comment (GalateaTom) [reply src/main/java/org/galatea/starter/entrypoint/BaseRestController.java]: Change made

File: src/main/java/org/galatea/starter/entrypoint/BaseRestController.java (hunk 3 above)
Comment (GalateaTom) [reply src/main/java/org/galatea/starter/entrypoint/BaseRestController.java]: Change made

**Review by GalateaEric — APPROVED** [review]

**Pull request merged**